core/Plugin.php

```diff
@@ -451,6 +451,19 @@ public static function getPluginNameFromNamespace($namespaceOrClassName)
     }
 
     /**
+     * Override this method in your plugin class if you want your plugin to be loaded during tracking.
+     *
+     * Note: If you define your own dimension or handle a tracker event, your plugin will automatically
+     * be detected as a tracker plugin.
+     *
+     * @return bool
+     */
+    public function isTrackerPlugin()
```

Comment:
I'd prefer not to, I don't like the 'component' approach where we do our own object creation, instead of using the more standard DI, and we dictate how plugin developers must develop. It seems very non-standard and a hack specific to PHP where you can't go through all available classes (due to autloading).